src/components/views/context_menus/StatusMessageContextMenu.js

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { _t } from '../../../languageHandler';
+import MatrixClientPeg from '../../../MatrixClientPeg';
+import AccessibleButton from '../elements/AccessibleButton';
+import classNames from 'classnames';
+
+export default class StatusMessageContextMenu extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this._onClearClick = this._onClearClick.bind(this);
```

Comment:
See the [style guide](https://github.com/matrix-org/matrix-react-sdk/blob/develop/code_style.md#react) for an approach that avoids needing to bind handlers like this.

Reply:
I'll still change it, but I really don't like the approach suggested :/

Reply:
Ah okay, sorry about that. It's possible we made the style guide change too quickly, as I believe only @ara4n and I discussed it. Code style can certainly be contentious, sometimes with opposing views for certain topics. We may want to use a different process for reviewing code style changes, so that we ensure more people on the team can voice their opinion if they wish to. Any thoughts about this?

Reply:
For now I think it's fine to follow the code style. Just voicing my opinion now so that if/when the code style comes into question again it's not a surprise to people. We probably don't need to iterate on the code style right now anyways, as the part I disagree with doesn't make a difference on maintainability or readability.